Ancestry Moogle (Race change service)

sruon opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Adds Ancestry Moogle NPC for race changes. 
  - Requires a charvar to be set up through an unspecified process for the NPC to respond.
  - A command script !racechange is provided, to set the charvars
- Adds a handful of supporting Lua bindings 
- Adds DNC AF swaps capability to the Moogle

- Documents unimplemented Item Depository NPC to warn about items that may need to be swapped
- Adds a couple enums for races, faces and character sizes.
- Adds some sanity checks in the login process

Capture
https://drive.google.com/file/d/1o6_9-Wk3iOMZVmNvP1S1HTm0T25Ou6pD/view?usp=sharing
https://www.youtube.com/watch?v=i5IXKNlQn_g

Trade capture
https://drive.google.com/file/d/1ZKlvUcCh-f8qSdnfSs8cWeU_avTC8kJm/view?usp=sharing

## Steps to test these changes

!racechange
!pos 51 8.5 -225 236

To repeat:
!racechange

<!-- Clear and detailed steps to test your changes here -->

sruon committed:
Lobby chr_info_sub2 race_change

Co-Authored-By: atom0s <atom0s@users.noreply.github.com>

sruon committed:
Race Change

zach2good commented on src/map/utils/charutils.cpp:
```diff
@@ -7496,4 +7503,33 @@ namespace charutils
+            return false;
+        }
+
+        if (newRace < static_cast<uint8>(CharRace::HumeMale) ||
```
I _believe_ you can take the enums as the arguments to this function and sol will break them down to their underlying types during binding, so you don't have to do all this casting. I'm not sure I've done the work for that on the way into the db though. I'm happy either way, might be worth looking into though

zach2good reviewed (approved):
This all looks extremely reasonable 👍

sruon commented:
- Changed some arguments to enum classes
- Captured and implemented the proper trade process for DNC AFs
- Added and tested NPCs in Windurst and San d'Oria
- Added a custom command script !racechange that toggles eligibility for the service
- Made charutils::raceChange unequip race/gender specific gear
